### DISCUSSION TITLE
Does srsRAN Project support X310?

### DISCUSSION THREAD

**Discussion by pengnist** [srsran/srsRAN_Project#228] [General Help]
We are exploring using X310 for gNB and ran into this error:

![image](https://github.com/srsran/srsRAN_Project/assets/71417513/f8254297-9425-4d3f-8d69-74b1435f340b)

We have an external octoclock that provides 10MHz reference clock and 1PPS to X310, and we added time_source = external and clock_source = external to the yaml file:

![image](https://github.com/srsran/srsRAN_Project/assets/71417513/8577bd10-f497-4713-8172-ec92eb7d6a24)

Does srsRAN Project support X310 for gNB radio? If so, is there any specific configuration we should make to the yaml?

Thanks

**Comment by robertfalkenberg** [comment]
Hi @pengnist, thanks for reporting this. Yes, the X310 is supported. I also get this warning, but so far the gNB keeps working normally.

**Reply by pengnist** [reply]
Hi @robertfalkenberg, thank you for confirming that X310 is supported. Can I ask if you have a recommended setup of the connections and tx/rx gain configurations? We are currently using SMA cable connection between gNB and UE, and the attenuation between each TX and RX is 60dB.

**Reply by ismagom** [reply]
with 60 dB attenuation better you transmit high power like tx_gain=30 and rx_gain 20

regards

**Reply by pengnist** [reply]
Thank you! I was able to make gnb work with X310, when the srsue uses B210. However, the connection cannot be established when srsue uses X310. If you have a working pair with X310s, do you mind sharing the conf file for your srsue?

The [rf] section of my ue.conf is:
```
freq_offset = 0
tx_gain = 30
rx_gain = 5
srate = 11.52e6
nof_antennas = 1

device_name = UHD
device_args = type=x300,addr=192.168.13.2,dboard_clock_rate=11.52e6,time_source=external,clock_source=external
time_adv_nsamples = auto
```
I also tried changing the time_adv_nsamples from 50 to 350 with no luck.

Please kindly let me know if any specific config needs to be applied. Thanks

**Reply by pgawlowicz** [reply]
Could you add the following to the UE config and see whether it works with x310?

```
[expert]
lte_sample_rates = true
```

**Reply by oliverxsch** [reply]
If having issues staying connected using two USRP X310, see my https://github.com/srsran/srsRAN_Project/issues/343, and this line lead to it not connecting at all.

**Reply by pgawlowicz** [reply]
@pengnist please check https://github.com/srsran/srsRAN_Project/issues/343
